src/frontend/contexts/LocationStoreContext.tsx

```diff
@@ -0,0 +1,125 @@
+import React, {createContext, useContext, useState} from 'react';
+import {
+  Accuracy,
+  LocationObject,
+  LocationSubscription,
+  getProviderStatusAsync,
+  useForegroundPermissions,
+  watchPositionAsync,
+} from 'expo-location';
+
+const LocationStoreContext = createContext<LocationStore | null>(null);
+
+// TODO: Return undefined when provider status changes to unavailable
+export function LocationStoreProvider({children}: {children: React.ReactNode}) {
+  const [store] = useState(() => new LocationStore());
+
+  const [locationPermissions] = useForegroundPermissions();
+
+  if (locationPermissions?.granted) {
```

Comment:
Suggestion: if you move this state into the LocationStore, then you can remove the restriction about only calling `init` after checking permissions, e.g. `store.setPermissions(locationPermissions)` - then the store can handle what it needs to do based on the permissions, and we can include the location permissions state in the LocationStore state, so it's clearer why the location might be null.